samples/sensor/env_sensing/src/main.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/sensor.h>
+#include <stdio.h>
+#include <sys/util.h>
+
+#include "disp.h"
+
+const struct device *display;
+const struct device *sensor_dev;
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(main);
+
+
+#if CONFIG_LOG
+#define sensor_log(...) LOG_INF(__VA_ARGS__)
+#else
+#define sensor_log(fmt, ...) printk(fmt, ##__VA_ARGS__)
+#endif
+
+static void process_sample(void)
+{
+	struct sensor_value gas, temp, hum, press;
+
+	if (sensor_sample_fetch(sensor_dev) < 0) {
+		LOG_ERR("Sensor sample update error");
+		return;
+	}
+#if CONFIG_HAS_TEMPERATURE
+	if (sensor_channel_get(sensor_dev, SENSOR_CHAN_AMBIENT_TEMP, &temp) < 0) {
+		LOG_ERR("Cannot read temperature channel");
+		return;
+	}
+#endif
+#if CONFIG_HAS_HUMIDITY
+	if (sensor_channel_get(sensor_dev, SENSOR_CHAN_HUMIDITY, &hum) < 0) {
+		LOG_ERR("Cannot read humidity channel");
+		return;
+	}
+#endif
+#if CONFIG_HAS_PRESSURE
+	if (sensor_channel_get(sensor_dev, SENSOR_CHAN_PRESS, &press) < 0) {
+		LOG_ERR("Cannot read pressure channel");
+		return;
+	}
+#endif
+#if CONFIG_HAS_GAS
+	if (sensor_channel_get(sensor_dev, SENSOR_CHAN_GAS_RES, &gas) < 0) {
+		LOG_ERR("Cannot read gas channel");
+		return;
+	}
+#endif
+
+
+#ifndef CONFIG_SAMPLE_DISPLAY_NONE
+	update_display(temp, hum, press, gas);
+#endif
+
+	if (IS_ENABLED(CONFIG_HAS_TEMPERATURE)) {
+		sensor_log("Temperature: %.1f C\n", sensor_value_to_double(&temp));
+	}
+	if (IS_ENABLED(CONFIG_HAS_HUMIDITY)) {
+		sensor_log("Relative Humidity: %.1f%%\n", sensor_value_to_double(&hum));
+	}
+	if (IS_ENABLED(CONFIG_HAS_PRESSURE)) {
+		sensor_log("Pressure: %.2f kPa", sensor_value_to_double(&press));
+	}
+	if (IS_ENABLED(CONFIG_HAS_GAS)) {
+		sensor_log("Gas: %.2f", sensor_value_to_double(&gas));
+	}
+}
+
+static void handler_(const struct device *dev,
+			   struct sensor_trigger *trig)
+{
+	process_sample();
+}
+
+
+void main(void)
+{
+
+#if defined(CONFIG_HTS221)
```

Comment:
+1 to remove all the duplicate samples but this PR just shift the issue to the one big sample, look at this #elfi tree below, this will continue to grow. The same for all the "config SAMPLE_SENSOR_*.

Reply:
this code was already removed in the latest patch and simplified even more. Still trying to find a more flexible solution with Kconfigs and DTS, but this is a price we have to pay to make this work regardless of board and sensor, we just need to make it suck less. Open for suggestions.

Reply:
Let us use labels and device name, and prefix it with device type, like you did it for I2C shell, or as I suggested in  #21709, :smile: duck and run away